pilot/pkg/networking/v1alpha3/cluster.go

```diff
@@ -60,36 +65,38 @@ func buildOutboundClusters(env model.Environment, services []*model.Service) []*
 	clusters := make([]*v2.Cluster, 0)
 	for _, service := range services {
 		config := env.DestinationRule(service.Hostname, "")
-		if config != nil {
-			destinationRule := config.Spec.(*networking.DestinationRule)
-			for _, port := range service.Ports {
-				hosts := buildClusterHosts(env, service, port)
-
-				// create default cluster
-				cluster := &v2.Cluster{
-					Name:  model.BuildSubsetKey(model.TrafficDirectionOutbound, "", service.Hostname, port),
-					Type:  convertResolution(service.Resolution),
-					Hosts: hosts,
-				}
-				applyTrafficPolicy(cluster, destinationRule.TrafficPolicy)
-				clusters = append(clusters, cluster)
+		for _, port := range service.Ports {
```

Comment:
For future PR: please try to untangle long method (refactor to smaller ones). Easier for testing, and will make it easier to switch the data structs and improve caching.